### ISSUE TITLE
exits when script errors

### ISSUE THREAD

**Issue by Konfekt** [ikysil/run-parts#8]
In contrast to https://github.com/wolfbox/run-parts this script exits at some point when a script errors out, but it is not yet clearly reproducibly. Sorry for being so vague; I'd be curious to know about possible reasons.

**Comment by ikysil** [comment]
> ... this script exits at some point when a script errors out, ...

I would like to clarify the statement above - shall I read it as
> ... **the run-parts** script exits at some point when a **child script** errors out,

?

**Comment by Konfekt** [comment]
Yes, again, I am sorry for the vague description. I used it for my monthly [anacron jobs](https://konfekt.github.io/blog/2016/12/11/sane-cron-setup#quick-start) and it kept aborting after around five of twenty scripts. Somehow the alternative one did not. I got interested in this version as the latter did not seem to work with [cronic](http://habilis.net/cronic/), but my understanding is limited if run-parts and cronic can play well together.

**Comment by Konfekt** [comment]
By the way, the description of `run-parts`

```
       If neither the --lsbsysinit option nor the --regex option is given  then  the  names  must
       consist  entirely of ASCII upper- and lower-case letters, ASCII digits, ASCII underscores,
       and ASCII minus-hyphens.
```

 reads as if it does not run scripts with a file extension such as `.sh`, but they are listed and run (partially here). Maybe it is understood that the name does not include the file extension.
